### PR TITLE
Add debug level logs and sync test to sim tests

### LINKED ISSUE #4275: Sim tests overhaul
**Background**

_Sim test = running a full devnet for some time_

Sim tests are crucial to test blockchain nodes due to their complexity. Unit tests or e2e test never capture the complexity of the system when it's running at full capacity. Even then running a full devnet in CI is not sufficient to capture all bugs since some only appear under certain network size and / or conditions.

The scope and goals of sim tests in this repo should be:
- [x] Run small devnet for a relatively short period of time.
- [x] Capture a large amount of metrics to ensure good performance on all fronts.
- [x] Do so every commit, or at least on every commit of the main branch

Sim tests are (and will be) complemented by:
- [x] ~~Kurtosis testing: Run small devnets for short time but with all other client combinations. Medium level of assertions~~ Moved to #4772
- [x] ~~Hive testing: Run small devnets for short time but with unusual network conditions. Low level of assertions https://github.com/ChainSafe/lodestar/issues/4240~~ Moved to #4773
- [x] Deployment on Prater (beta testing): Run big devnet for long periods of time, but only infrequently before releases

**TODO**

Current sim test implementation is very old an subpar. It could be structured in a way that more modular, mantainable, and stable. However the first important step is to add assertions to cover the missing tick in the scope outline above:

- [x] Add proper assertions to sim tests https://github.com/ChainSafe/lodestar/issues/4276
- [x] Migrate leftover sim tests from e2e, to make e2e tests more stable and fast
- [x] Consolidate sim tests into less runs: do lightclient, finality, sync tests in one run https://github.com/ChainSafe/lodestar/issues/3931
- [x] Enable multi-node multi-process testing
  - Attempt at implementing https://github.com/ChainSafe/lodestar/pull/2575
  - Blocking issue https://github.com/ChainSafe/lodestar/issues/2542
- [x] ~~If expensive at this point, run in self-deployed GA runner.~~ **EDIT (nov 11th)**: keep the run light to work in Github runners

**Progressive improvements**
- [x] Log std to file, disable log to file. Enable DEBUG=* for Lodestar
- [x] Run Geth and Nethermind and test going through the merge, until finalizing merge block + 2 epochs
- [x] Ensure all nodes are connected to each other https://github.com/ChainSafe/lodestar/issues/4645
- [x] Once nodes reach untilEpoch, keep the chain running and spawn extra nodes to test syncing
  - [x] Start 1 node from genesis and range sync to head
  - [x] Start 1 node with checkpoint sync and sync to head  
- [x] Update the test timeout to be calculated [deterministically](https://github.com/ChainSafe/lodestar/pull/4673#discussion_r1007250212)
- [x] Disable the file logging for lodestar by explicitly setting it to `none`
- [x] Update the sync tests approach by just testing the sync behavior and not waiting for a particular epoch.

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Add a new node to sim test network and test the syncing from genesis. 

Covered following points: 

- [x] Log std to file, disable log to file. Enable DEBUG=* for Lodestar
- [x] Once nodes reach untilEpoch, keep the chain running and spawn extra nodes to test syncing
  - [x] Start 1 node from genesis and range sync to head
  - [x] Start 1 node with checkpoint sync and sync to head  
- [x] Disable the file logging for lodestar by explicitly setting it to none

Partially Resolves #4275

**Steps to test or reproduce**

Run sim tests.

### CODE REVIEW

File: packages/cli/test/utils/simulation/cl_clients/lodestar.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+import {Keystore} from "@chainsafe/bls-keystore";
+import {nodeUtils} from "@lodestar/beacon-node/node";
+import {LogLevel} from "@lodestar/utils";
+import {IBeaconArgs} from "../../../../src/cmds/beacon/options.js";
+import {IValidatorCliArgs} from "../../../../src/cmds/validator/options.js";
+import {IGlobalArgs} from "../../../../src/options/globalOptions.js";
+import {JobOptions, CLClientGenerator, CLClientOptions, Runner, RunnerType, Job} from "../interfaces.js";
+import {callHttp, LODESTAR_BINARY_PATH} from "../utils.js";
+
+export const generateLodeStarBeaconNode: CLClientGenerator = (opts: CLClientOptions, runner: Runner): Job => {
```
Comment (wemeetagain) [inline packages/cli/test/utils/simulation/cl_clients/lodestar.ts]: small thing, but we usually treat lodestar as a single word.

Lodestar
lodestar

not 

LodeStar
lodeStar
lode_star

File: packages/cli/test/simulation/simulation.test.ts [hunk 2]
```diff
@@ -62,14 +62,13 @@ for (const {beaconNodes, validatorClients, validatorsPerClient} of nodeCases) {
...
-      genesisSlotsDelay: (SLOTS_PER_EPOCH * runTill + 50) * testCases + 30,
+      genesisSlotsDelay: SLOTS_PER_EPOCH * 2,
       bellatrixEpoch,
       logFilesDir: join(logFilesDir, testIdStr),
     });
-    testCases += 1;
 
     describe(`simulation test - ${testIdStr}`, function () {
-      this.timeout("5m");
+      this.timeout("10m");
```
Comment (dapplion) [inline packages/cli/test/simulation/simulation.test.ts:71]: Can you add an issue to compute this timeout dynamically in a number that makes sense given the test requirements? i.e. expected duration time + 10%

File: packages/cli/test/utils/simulation/ExternalSignerServer.ts [hunk 3]
```diff
@@ -46,28 +43,34 @@ export class ExternalSignerServer {
       const pubkeyHex: string = req.params.identifier;
       const signingRootHex: string = req.body.signingRoot;
 
-      const secretKey = secretKeyMap.get(pubkeyHex);
+      const secretKey = this.secretKeyMap.get(pubkeyHex);
       if (!secretKey) {
+        console.log([...this.secretKeyMap.keys()].join("\n"));
```
Comment (dapplion) [inline packages/cli/test/utils/simulation/ExternalSignerServer.ts]: Is this console.log necessary? Keys are deterministic

File: packages/cli/test/utils/simulation/utils.ts [hunk 4]
```diff
@@ -90,8 +52,40 @@ export const getForkName = (epoch: Epoch, params: SimulationParams): ForkName =>
...
 };
 
+export const callHttp = async (
+  url: string,
+  data?: unknown
+): Promise<{status: number; data: Record<string, unknown>}> => {
+  const opts = parse(url);
+
+  return new Promise((resolve, reject) => {
+    const req = request({...opts, headers: {"Content-Type": "application/json"}}, function (res) {
```
Comment (dapplion) [inline packages/cli/test/utils/simulation/utils.ts]: lodestar/cli package already has dependencies on some http client libraries, use those

**Review by dapplion — DISMISSED** [review]
Some minor comments, otherwise good to merge

File: packages/cli/test/simulation/simulation.test.ts (hunk 2 above)
Comment (nazarhussain) [reply packages/cli/test/simulation/simulation.test.ts:71]: Added it as task to the main issue.

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**